trace/sns/publisher.go

```diff
@@ -57,17 +57,15 @@ func (p TracedPublisher) Publish(ctx context.Context, msg Message) (messageID st
 
 	out, err := p.api.PublishWithContext(ctx, msg.input)
 
+	defer trace.SpanComplete(span, err)
```

Comment:
No need to defer here.
```suggestion
	trace.SpanComplete(span, err)
```